src/libtiled/tilelayer.cpp

```diff
@@ -594,8 +659,31 @@ TileLayer *TileLayer::clone() const
 TileLayer *TileLayer::initializeClone(TileLayer *clone) const
 {
     Layer::initializeClone(clone);
-    clone->mGrid = mGrid;
+    QHashIterator<QPoint, Chunk* > it(mChunks);
+    while (it.hasNext()) {
+        it.next();
+        for (int x = 0; x < CHUNK_SIZE; ++x) {
+            for (int y = 0;y < CHUNK_SIZE; ++y) {
+                if (!contains(it.key().x() * CHUNK_SIZE + x,
+                              it.key().y() * CHUNK_SIZE + y))
+                    continue;
+
+                clone->setCell(it.key().x() * CHUNK_SIZE + x,
+                               it.key().y() * CHUNK_SIZE + y,
+                               it.value()->cellAt(x, y));
```

Comment:
Actually this is a needlessly slow way of cloning. You should just copy over each chunk:
```c++
    QHashIterator<QPoint, Chunk* > it(mChunks);
    while (it.hasNext()) {
        it.next();
        clone->mChunks.insert(it.key(), new Chunk(*it.value()));
    }
```